File: src/Other/MicroPython/prototype.py
def clone_profile(profile):
    return {
        "name": profile["name"],
        "features": list(profile["features"]),
    }


def describe(profile):
    return "{}: {}".format(profile["name"], ",".join(profile["features"]))


original = {"name": "orders", "features": ["metrics"]}
canary = clone_profile(original)
canary["name"] = "orders-canary"
canary["features"].append("tracing")

print("original=" + describe(original))
print("clone=" + describe(canary))

File: src/Scripting/Perl/prototype.pl
use strict;
use warnings;

sub new_profile {
    my ($name, @features) = @_;
    return {
        name => $name,
        features => [@features],
    };
}

sub clone_profile {
    my ($source) = @_;
    return {
        name => $source->{name},
        features => [@{$source->{features}}],
    };
}

sub describe {
    my ($profile) = @_;
    return $profile->{name} . ': ' . join(',', @{$profile->{features}});
}

my $original = new_profile('orders', 'metrics');
my $canary = clone_profile($original);
$canary->{name} = 'orders-canary';
push @{$canary->{features}}, 'tracing';

print 'original=' . describe($original) . "\n";
print 'clone=' . describe($canary) . "\n";
